containers/podman-desktop-extension-ai-lab
# feat: show warning if no machine is running or there are not enough resources (#227)

### What does this PR do?

It shows a warning banner if the running podman machine does not have enough resources for the model selected by the user or if there is no running machine.

This needs https://github.com/containers/podman-desktop/pull/6733 to work

This PR only covers the inference server page to make the review easier. Going to open a new PR for the recipe page once it is merged

### Screenshot / video of UI

This gif shows if you can edit the machine and if there is no running machine
![warning_banner_1](https://github.com/containers/podman-desktop-extension-ai-lab/assets/49404737/61024d8d-5475-4144-8d01-fb5d964c5192)


This is the message if the edit action is not enabled for the system (like WSL)
![image](https://github.com/containers/podman-desktop-extension-ai-lab/assets/49404737/147886f5-a29b-4e74-9784-1b5a4bfc0e53)


### What issues does this PR fix or reference?

Fixes #873 

### How to test this PR?

1. if you are on WSL you need to hack the result received from the system as it takes the resources from the host so you have to load the machine a lot to see the warning message
2. on MAC by creating a machine with few resources you should see the banner
